crates/api_common/src/site.rs

```diff
@@ -316,9 +323,48 @@ pub struct MyUserInfo {
 #[cfg_attr(feature = "full", ts(export))]
 /// A list of federated instances.
 pub struct FederatedInstances {
-  pub linked: Vec<Instance>,
-  pub allowed: Vec<Instance>,
-  pub blocked: Vec<Instance>,
+  pub linked: Vec<InstanceWithFederationState>,
+  pub allowed: Vec<InstanceWithFederationState>,
+  pub blocked: Vec<InstanceWithFederationState>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[cfg_attr(feature = "full", derive(TS))]
+#[cfg_attr(feature = "full", ts(export))]
+pub struct ReadableFederationState {
+  /// the last successfully sent activity id
+  last_successful_id: ActivityId,
```

Comment:
Last successful send time seems more useful.

Reply:
It would yes, but one issue there is that we (currently) only know the last id and could get the timestamp of that id via the sent_activities table only if it has not been pruned yet. Also even if everything is functioning correctly the timestamp will be 10-120 seconds out of date (because it's not instantly written to the DB) so it might be a bit confusing.

So I could

1. leave it as is,
2. copy the published column of the last successful_id into the federation_queue_state table
3. join the sent_activities table into the added api request, fetching the published column of the activity directly if it is still known

Reply:
The activity id seems completely useless on its own. Im not sure what the differences of your option 2 and 3 would be. Either one is fine as long as it gives a timestamp thats halfway accurate. No one will care if its a few minutes wrong, what matters is if federation with an instance is broken for some hours or days.

Reply:
It's useful only for relative comparisons. You can calculate the difference between activity ids returned in this call and see that relative to up-to-date instance X, instance Y is 1000 behind, instance Z is 100 behind, etc

The difference between 2 and 3 is that if an instance is down longer than the prune duration of sent activities (the scheduled job) , (3) will show null. 

I think I'll implement (2)